torchtune/modules/attention_utils.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import Callable, List, Optional, Union
+
+import torch
+
+from torch import nn
+from torchtune.utils._version import torch_version_ge
+from torchtune.utils.logging import get_logger, log_once
+
+# We can only use flex attention / BlockMask if torch version >= 2.5.0 and GPU is Turing / SM75 and above
+_SUPPORTS_FLEX_ATTENTION = torch_version_ge(
+    "2.5.0"
+) and torch.cuda.get_device_capability() >= (7, 5)
```

Comment:
This will error if we're running purely on CPU I think

Reply:
can add `torch.cuda.is_available()`